tests/unit/recommenders/evaluation/test_spark_evaluation.py

```diff
@@ -201,83 +69,83 @@ def test_init_spark_rating_eval(spark_data):
 
 
 @pytest.mark.spark
-def test_spark_rmse(spark_data, target_metrics):
+def test_spark_rmse(spark_data):
     df_true, df_pred = spark_data
 
-    evaluator1 = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
-    assert evaluator1.rmse() == 0
+    evaluator = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
+    assert evaluator.rmse() == 0
 
-    evaluator2 = SparkRatingEvaluation(df_true, df_pred)
-    assert evaluator2.rmse() == target_metrics["rmse"]
+    evaluator = SparkRatingEvaluation(df_true, df_pred)
+    assert evaluator.rmse() == pytest.approx(7.254309, TOL)
 
 
 @pytest.mark.spark
-def test_spark_mae(spark_data, target_metrics):
+def test_spark_mae(spark_data):
     df_true, df_pred = spark_data
 
-    evaluator1 = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
-    assert evaluator1.mae() == 0
+    evaluator = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
+    assert evaluator.mae() == 0
 
-    evaluator2 = SparkRatingEvaluation(df_true, df_pred)
-    assert evaluator2.mae() == target_metrics["mae"]
+    evaluator = SparkRatingEvaluation(df_true, df_pred)
+    assert evaluator.mae() == pytest.approx(6.375, TOL)
 
 
 @pytest.mark.spark
-def test_spark_rsquared(spark_data, target_metrics):
+def test_spark_rsquared(spark_data):
     df_true, df_pred = spark_data
 
-    evaluator1 = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
-    assert evaluator1.rsquared() == pytest.approx(1.0, TOL)
+    evaluator = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
+    assert evaluator.rsquared() == pytest.approx(1.0, TOL)
 
-    evaluator2 = SparkRatingEvaluation(df_true, df_pred)
-    assert evaluator2.rsquared() == target_metrics["rsquared"]
+    evaluator = SparkRatingEvaluation(df_true, df_pred)
+    assert evaluator.rsquared() == pytest.approx(-31.699029, TOL)
 
 
 @pytest.mark.spark
-def test_spark_exp_var(spark_data, target_metrics):
+def test_spark_exp_var(spark_data):
     df_true, df_pred = spark_data
 
-    evaluator1 = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
-    assert evaluator1.exp_var() == pytest.approx(1.0, TOL)
+    evaluator = SparkRatingEvaluation(df_true, df_true, col_prediction="rating")
+    assert evaluator.exp_var() == pytest.approx(1.0, TOL)
 
-    evaluator2 = SparkRatingEvaluation(df_true, df_pred)
-    assert evaluator2.exp_var() == target_metrics["exp_var"]
+    evaluator = SparkRatingEvaluation(df_true, df_pred)
+    assert evaluator.exp_var() == pytest.approx(-6.4466, 0.01)
```

Comment:
Same question here. So both spark and python evaluations now give slightly different results. Maybe something changed in the input dataframes?

Reply:
this should use the same `TOL` for the consistency.